### PR TITLE
refactor: clean up package metadata

### PR DESCRIPTION
**Pull request by weihanglo**
Like #12352 but for `homepage` and `repository`.

Inspired by <https://github.com/rust-lang/cargo/pull/13178#discussion_r1429394873>.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -126,8 +128,8 @@ version = "0.77.0"
 edition.workspace = true
 license.workspace = true
 rust-version.workspace = true
-homepage = "https://crates.io"
-repository = "https://github.com/rust-lang/cargo"
+homepage.workspace = true
```
Comment (weihanglo) [inline Cargo.toml:131]: Changed from <https://crates.io> to https://github.com/rust-lang/cargo>. Should be acceptable?

Comment (weihanglo) [reply Cargo.toml:131]: Good idea. Are we going to link to The Cargo Book for all subcrates or just `cargo` main crate?

Comment (epage) [reply Cargo.toml:131]: I think that only makes sense for `cargo`

**Pull request merged**